src/NetworkSettings.cpp

```diff
@@ -110,6 +111,31 @@ void NetworkSettingsClass::raiseEvent(network_event event)
     }
 }
 
+void NetworkSettingsClass::handleMDNS()
+{
+    bool mdnsEnabled = Configuration.get().Mdns_Enabled;
+
+    if (lastMdnsEnabled != mdnsEnabled) {
+        MDNS.end();
+
+        if (mdnsEnabled) {
+            if (MDNS.begin(getHostname().c_str())) {
```

Comment:
There is no need to call `.c_str()` as the begin method uses String internaly anyways.